src/main/AllocatorSuperblock.cpp

```diff
@@ -0,0 +1,297 @@
+/*
+ *  R : A Computer Language for Statistical Data Analysis
+ *  Copyright (C) 1995, 1996  Robert Gentleman and Ross Ihaka
+ *  Copyright (C) 1998-2007   The R Development Core Team.
+ *  Copyright (C) 2008-2014  Andrew R. Runnalls.
+ *  Copyright (C) 2014 and onwards the Rho Project Authors.
+ *
+ *  Rho is not part of the R project, and bugs and other issues should
+ *  not be reported via r-bugs or other R project channels; instead refer
+ *  to the Rho website.
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, a copy is available at
+ *  https://www.R-project.org/Licenses/
+ */
+
+#include <assert.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#include <cstdint>
+#include <cstdlib>
+#include <functional>
+#include <limits>
+#include <map>
+
+#include "rho/AddressSanitizer.hpp"
+#include "rho/AllocationTable.hpp"
+#include "rho/AllocatorSuperblock.hpp"
+
+unsigned rho::AllocatorSuperblock::s_size_class[] = {
+       0,      0,      0,      0,  4 * 8,  5 * 8,  6 * 8,  7 * 8,
+   8 * 8,  9 * 8, 10 * 8, 11 * 8, 12 * 8, 13 * 8, 14 * 8, 15 * 8,
+  16 * 8, 17 * 8, 18 * 8, 19 * 8, 20 * 8, 21 * 8, 22 * 8, 23 * 8,
+  24 * 8, 25 * 8, 26 * 8, 27 * 8, 28 * 8, 29 * 8, 30 * 8, 31 * 8,
+  32 * 8,
+  1 << 0, 1 << 1, 1 << 2, 1 << 3, 1 << 4, 1 << 5, 1 << 6, 1 << 7,
+  1 << 8, 1 << 9, 1 <<10, 1 <<11, 1 <<12, 1 <<13, 1 <<14, 1 <<15,
+  1 <<16, 1 <<17,
+};
+
+// Arena for small-object superblocks:
+void* superblock_arena = nullptr;
```

Comment:
AFAICT, this isn't used anywhere outside of allocateArena(), so it can be turned into a local variable.